### ISSUE TITLE
Bug: ESLint with typescript plugin does not initialize correctly after going through @eslint/config

### ISSUE THREAD

**Issue by nickph7** [eslint/eslint#16353]
### Environment

Node version: v14.19.0
npm version: 6.14.16
Local ESLint version: 8.24.0
Global ESLint version: None
Operating System: Ubuntu 20.04.5 LTS
Typescript version: 4.8.3 (installed globally)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

After initializing eslint on an empty project with typescript option, eslint complains that @typescript-eslint declared in .eslintrc.json is missing module typescript.

Detail step of what I did to reproduce the problem:

- Created a new empty directory and ran `npm init --yes`
- Added eslint using `npm init @eslint/config`
- Selected the following options: problems, esm, no framework, with typescript, browser and node, JSON file
- Created a file index.ts with `console.log("Hello World")
- Ran npx eslint index.ts

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "browser": true,
        "es2021": true,
        "node": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "overrides": [
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
}
```

```ts
console.log("Hello World")
```

### What did you expect to happen?

Eslint is installed correctly from the @eslint/config process and lint the file without error.

### What actually happened?

Eslint complains it cannot find the typescript module. 

```
npx eslint index.ts

Oops! Something went wrong! :(

ESLint: 8.24.0

Error: Failed to load plugin '@typescript-eslint' declared in '.eslintrc.json': Cannot find module 'typescript'
Require stack:
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/util/astUtils.js
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/util/index.js
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/rules/adjacent-overload-signatures.js
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
- /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@eslint/eslintrc/dist/eslintrc.cjs
Referenced from: /home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/.eslintrc.json
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at Object.<anonymous> (/home/nicoloprudo/codews/_TEMP-PROJECTS/eslint-install-test/node_modules/@typescript-eslint/eslint-plugin/dist/util/astUtils.js:30:25)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Unsure if it should be in this repo, because of an error related to @config/init, or typescript repo, because the plugin cannot find the module

**Comment by nickph7** [comment]
I found that global install of typescript is unsupported
